### DISCUSSION TITLE
Get Python2

### DISCUSSION THREAD

**Discussion by HugoGarcia22** [Homebrew/discussions#915] [Everyday usage]
How can I install python@2.7 on the latest version of Homebrew?
If I do brew search python the version python@2 is not available.
Or can I install other version of Homebrew that contains Python2?
I need python2.7. specifically to run a program
Thanks.

**Comment by SMillerDev** [comment]
You can not, Python stopped support for Python 2 in Februari 2020 and so did Homebrew.

If you're on macOS you can use the system python if you really need a python 2. But since this is unsupported software I would recommend moving to python 3 instead.

**Reply by HugoGarcia22** [reply]
Thanks, I apreciate it. But I really need python2.7. specifically to run CLASS (Cosmic Linear Anisotropy Solving System)

**Reply by SMillerDev** [reply]
There is some work being done in https://github.com/lesgourg/class_public/pulls to make it Py3 compliant. But unfortunately Homebrew can't help you with python 2.

**Reply by danielnachun** [reply]
You can also get python2 from Miniconda: https://docs.conda.io/en/latest/miniconda.html.  That's what I've had to use for the rare case where I need python2.

**Reply by marcink** [reply]
MacOS just eliminated python2 from system install.  How can we get python@2 back as an available formula?

I've heard the argument that Python officially dropped support for Python 2, but that does not reflect reality of where python 2.7 is still used in hardened or semi-hardened deployed systems, like CentOS 7 that defaults to Python 2.  Developing code that must maintain backwards compatibility with 2.7 requires a local install.

**Reply by SMillerDev** [reply]
> How can we get python@2 back as an available formula?

You can compile it yourself as a formula but it will never come back officially.

> python 2.7 is still used in hardened or semi-hardened deployed systems, like CentOS 7

Unlike RedHat, Homebrew follows upstream directions for supporting software.

**Reply by danielnachun** [reply]
Although Python 2 is the default on CentOS 7, there's no reason the software running on it can't use Python 3.  

As I mentioned above, if you need Python 2 on macOS, it is available in Anaconda and will probably be there indefinitely.  It's not as convenient as using Homebrew, but it will work.
